modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -555,6 +864,37 @@ public void onFailure(Exception e) {
         });
     }
 
+    private void downsampleIndex(DownsampleAction.Request request, ActionListener<Void> listener) {
+        String sourceIndex = request.getSourceIndex();
+        String downsampleIndex = request.getTargetIndex();
+        logger.info("Data stream lifecycle issuing request to downsample index [{}] to index [{}]", sourceIndex, downsampleIndex);
+        client.execute(DownsampleAction.INSTANCE, request, new ActionListener<>() {
+            @Override
+            public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                logger.info("Data stream lifecycle successfully downsampled index [{}] to index [{}]", sourceIndex, downsampleIndex);
+                listener.onResponse(null);
+            }
```

Comment:
Same here, should we check the acknowledged response to make sure it's actually been acknowledged?

Reply:
Adding an assertion as it's always acknowledged